### DISCUSSION TITLE
The problem of lithium ion diffusivity input in different directions

### DISCUSSION THREAD

**Discussion by biaogxb** [idaholab/moose#21317] [Q&A Modules: Phase field] [answered]
I want to use the phase field model to simulate the growth of dendrites,and I have some problems when I input the anisotropic mobility of lithium ions.
I use ‘MatDiffusion’ to simulate the migration of lithium ions, as shown in the following code

` [./Diffusion1]` 
   ` type = MatDiffusion` 
   ` variable=w` 
   ` D_name=D` 
 ` [../]`

` [./Mobility_coefficient] ` 
   ` type = DerivativeParsedMaterial` 
    ` function = '-M0*(1-h)*cl*(1-h)'  
    ` f_name = D` 
     ` args = 'eta w'` 
    ` derivative_order = 1` 
    ` material_property_names = ' M0 cl:=D[f1,w] h'` 
  ` [../]`

Where M0 represents diffusion coefficient of li+ in electricity.At first I set a constant, but now I want to change M0 to the following form：
![image](https://user-images.githubusercontent.com/95063954/173985746-b393de89-7425-4b39-90bd-9a09d7d3b04b.png)
where, the diagonal components represent mobility coefficients in x and y directions.

In a word, I want to change the previous constant diffusion coefficient of li+ in electricity into the tensor as shown in the figure and use it to calculate ’D‘，finally I bring D into the diffusion equation. How can I achieve it？

![image](https://user-images.githubusercontent.com/95063954/174043651-2bddbc65-336e-4d39-b8ba-24ce915c2bbc.png)

 I add a ‘Mobility_tensor’ block  and then change the place where M0 is used to M directly.I don't know if this modification is correct.
 The code is as follows：
 ` [./Diffusion1] ` 
   ` type = MatAnisoDiffusion ` 
   ` diffusivity = D` 
   ` variable = w` 
   ` [../]`

`[./Mobility_tensor]` 
      ` type = ConstantAnisotropicMobility` 
      ` M_name = M` 
      ` tensor = '317.9 0 0 317.9'` 
    ` [../]`

 ` [./Mobility_coefficient]` 
    ` type = DerivativeParsedMaterial` 
    ` function = '-M*(1-h)*cl*(1-h)'  ` 
    ` f_name = D` 
    ` args = 'eta w'` 
    ` derivative_order = 1` 
    ` material_property_names = ' M cl:=D[f1,w] h'` 
  ` [../]`

 Can someone help me? Thank you！

**Comment by GiudGiud** [comment, marked as answer]
Hello

The derivative parsed material does not handle the matrix material property I think.
If you need a custom expression for an anisotropic diffusion coefficient, I think you will want to make a custom object. 
Unless there are other XXXAnisotropicMobility objects you can use

Guillaume

**Reply by biaogxb** [reply]
OK, thanks. I'll study this further
